### PR TITLE
SDIT-1528 Throw 403 for /api/bookings/offenderNo/{offenderNo} if user does not have access

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 1]
```diff
@@ -575,6 +575,10 @@ public void verifyBookingAccess(final Long bookingId, boolean accessDeniedError,
                 logClientUnauthorisedAccess(bookingId, rolesAllowed);
                 throw new AccessDeniedException(format("Client not authorised to access booking with id %d.", bookingId));
             }
+            if (accessDeniedError) {
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:578]: If the user does not have any caseloads, then we should still throw a 403

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 2]
```diff
@@ -599,6 +603,7 @@ private void logUserUnauthorisedAccess(final Long bookingId, final Set<String> a
         final Map<String, String> logMap = new HashMap<>();
         logMap.put("bookingId", bookingId.toString());
         logMap.put("bookingCaseload", bookingAgencyId);
+        logMap.put("clientId", authenticationFacade.getClientId());
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:606]: clientId now available in auth details

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 3]
```diff
@@ -177,7 +177,7 @@ public InmateDetail getOffenderBooking(
         @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
     @Operation(summary = "Offender detail.", description = "Offender detail.")
     @GetMapping("/offenderNo/{offenderNo}")
-    @VerifyOffenderAccess(overrideRoles = {"GLOBAL_SEARCH", "VIEW_PRISONER_DATA"})
+    @VerifyOffenderAccess(overrideRoles = {"GLOBAL_SEARCH", "VIEW_PRISONER_DATA"}, accessDeniedError = true)
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:180]: The majority of the current 404s (all except one) are from prisonstaffhub which follows the same path for all errors for this call (i.e. no change is required)

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**